quay-enterprise/setup-elb.md

```diff
@@ -0,0 +1,131 @@
+```
+
+Describe the `quay-ProxyProtocol-policy`  policy to ensure correctness. 
+
+```
+$ aws elb describe-load-balancer-policies --load-balancer-name quay-loadbalancer --policy-names quay-ProxyProtocol-policy
+```
+
+Output should match the following: 
+
+```
+{
+    "PolicyDescriptions": [
+        {
+            "PolicyName": "quay-ProxyProtocol-policy",
+            "PolicyAttributeDescriptions": [
+                {
+                    "AttributeName": "ProxyProtocol",
+                    "AttributeValue": "true"
+                }
+            ],
+            "PolicyTypeName": "ProxyProtocolPolicyType"
+        }
+    ]
+}
+```
+
+### Set the Proxy Protocol policy on instance port 8443
+
+```
+$ aws elb set-load-balancer-policies-for-backend-server --load-balancer-name quay-loadbalancer --instance-port 8443 --policy-names quay-ProxyProtocol-policy
+```
+
+### View configuration of ELB
+
+Confirm that the Proxy Protocol policy has  been set to on instance port 8443 by looking for the output of `BackendServerDescriptions` in the ELB description: 
+
+```
+$ aws elb describe-load-balancers --load-balancer-name quay-loadbalancer
+[...]
+{
+    "LoadBalancerDescriptions": [
+        {
+            ...
+            "BackendServerDescriptions": [
+                {
+                    "PolicyNames": [
+                        "quay-ProxyProtocol-policy"
+                    ],
+                    "InstancePort": 8443
+                    ]
+                }
+            ], 
+            ...
+        }
+    ]
+}
+```
+
+
+
+### Configure listeners to forward traffic to Quay Enterprise
+
+A TCP listener should be configured to route traffic from Load Balancer Port 443 to Instance Port 8443. This is most easily configured from the AWS console. 
+
+<img src="img/editlisteners.png" class="img-center" alt="Edit Listeners"/>
+
+When properly configured, the Listeners tab for the ELB should appear like so:
+
+<img src="img/listenerstab.png" class="img-center" alt="Listeners">
+
+### Add Health Check 
+
+After the listeners have been configured, the health checking endpoint needs to be configured to use the previously unused port 443. When properly configured, the Listeners tab for the ELB should appear like so:
+
+<img src="img/healthcheck.png" class="img-center" alt="Healthcheck">
+
+### Check Functionality 
+
+If the ELB with Proxy Protocol is functioning properly curling the `/v1/_ping` endpoint should return `true`. Replace ${LoadBalancer} with the A record of the quay-loadbalancer ELB..
+
+```
+$ curl -k https://${LoadBalancer}/v1/_ping
+true
+```
+
+`docker login` and `docker push` should be operational. 
+
+## Troubleshooting 
+
+[Security group](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-network-security.html) settings are often the culprit if the ELB is not resolving but the health check reports the instance as in service. The security group for the ELB should be open for inbound traffic from port `80`, `443`, and `8443`. 
```

Comment:
Add language that if the security group is misconfigured, it generally shows up as the connection hanging